### PR TITLE
Add id to pointer data to handle multi-touch screens

### PR DESCRIPTION
**Pull request by malaref**
Add touch identifier to pointer data to handle multi-touch screens
Ref: https://www.w3.org/TR/touch-events/#widl-Touch-identifier

### CODE REVIEW

File: packages/clarity-js/src/interaction/encode.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export default async function (type: Event, ts: number = null): Promise<void> {
                     tokens.push(pTarget.id);
                     tokens.push(entry.data.x);
                     tokens.push(entry.data.y);
+                    if (entry.data.id !== undefined) { tokens.push(entry.data.id); }
```
Comment (toby-walker) [inline packages/clarity-js/src/interaction/encode.ts:40]: When would this be undefined?

File: packages/clarity-js/src/interaction/pointer.ts [hunk 2]
```diff
@@ -57,9 +57,10 @@ function touch(event: Event, root: Node, evt: TouchEvent): void {
             let y = "clientY" in entry ? Math.round(entry["clientY"] + d.scrollTop) : null;
             x = x && frame ? x + Math.round(frame.offsetLeft) : x;
             y = y && frame ? y + Math.round(frame.offsetTop) : y;
+            const id = "identifier" in entry ? entry["identifier"] : undefined;
```
Comment (toby-walker) [inline packages/clarity-js/src/interaction/pointer.ts:60]: The server side is assuming id=0 is the primary pointer and the rest are != o. However there is no guarantee in teh spec of that behavior. We need to do the number ourselves and map from the arbitrary touch id to the simple numbering scheme.

Basic idea: 

* Keep a dictionary of id -> touch number
* When it's a touchstart event then add a new entry to the dictionary with the touch # assigned starting from 0.

Comment (malaref) [reply packages/clarity-js/src/interaction/pointer.ts:60]: The `identifier` is not arbitrary, it's 0-based and it resets when all the fingers are off the screen. It's not a part of the spec, but that's the implemented behaviour in browsers (tested on Chrome-based ones as well as Firefox).

File: packages/clarity-js/src/interaction/encode.ts (hunk 1 above)
Comment (malaref) [reply packages/clarity-js/src/interaction/encode.ts:40]: When it's a mouse (click) event, not touch.

File: packages/clarity-js/src/interaction/pointer.ts (hunk 2 above)
Comment (toby-walker) [reply packages/clarity-js/src/interaction/pointer.ts:60]: Can you please add a comment that capture the above (starts at zero and confirmed on the 3 major browse engines?

Comment (malaref) [reply packages/clarity-js/src/interaction/pointer.ts:64]: done

**Review by ender336 — APPROVED** [review]
:shipit:

**Review by jjshaws — APPROVED** [review]

**Pull request merged**